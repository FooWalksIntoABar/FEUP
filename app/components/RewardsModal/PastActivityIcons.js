import React from 'react'
import { View, Image } from 'react-native'
import PropTypes from 'prop-types'
import * as Animatable from 'react-native-animatable'
import styles from '../../styles/RewardModal.style'
import Images from '../../assets/images/images'

export class PastActivityIcons extends React.Component {
  filterActivities = (_, index, array) => {
    if (this.props.currentActivity + 8 > array.length) return index >= array.length - 8
    else return index >= this.props.currentActivity && index < this.props.currentActivity + 8
  }

  render () {
    let greyedOutIcons = this.props.activities
      .filter(this.filterActivities)
      .map((activity, index) => activity.status
        ? (<View key={index} style={styles.grayedOutActivityIcon} />)
        : (<View key={index} style={styles.grayedOutActivityIcon}>
          <Image style={activity.photo ? styles.grayedOutActivityPhoto : styles.grayedOutActivityImage} recizeMode={'cover'} source={activity.photo ? Images[activity.photo] : Images[activity.image]} />
        </View>)
      )

    let pastActivityIcons = this.props.activities
      .filter(this.filterActivities)
      .reduce((accumulator, activity, index) => {
        if (activity.status) {
          return {
            counter: accumulator.counter + 1,
            views: [...accumulator.views, (<Animatable.View key={index} animation={'zoomIn'} delay={600 + accumulator.counter * 1000} style={[styles.pastActivityIcon, { backgroundColor: activity.color }]}>
              <Image style={activity.photo ? styles.pastActivityPhoto : styles.pastActivityImage} recizeMode={'cover'} source={activity.photo ? Images[activity.photo] : Images[activity.image]} />
            </Animatable.View>)]
          }
        } else {
          return {
            counter: accumulator.counter,
            views: [...accumulator.views, (<View key={index} style={styles.pastActivitySpacer} />)]
          }
        }
      }, { counter: 0, views: [] }).views

    return (
      <View style={styles.pastActivitiesContainer}>
        <View style={styles.greyedOutActivitiesContainer}>
          {greyedOutIcons}
        </View>
        <View style={styles.completedActivitiesContainer}>
          {pastActivityIcons}
        </View>
      </View>
    )
  }
}

PastActivityIcons.propTypes = {
  activities: PropTypes.array.isRequired,
  currentActivity: PropTypes.number.isRequired
}
