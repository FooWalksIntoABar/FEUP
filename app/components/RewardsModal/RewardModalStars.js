import React from 'react'
import { View } from 'react-native'
import PropTypes from 'prop-types'

import * as Animatable from 'react-native-animatable'

import styles from '../../styles/RewardModal.style'
import Images from '../../assets/images/images'

export class RewardModalStars extends React.Component {
  constructor (props) {
    super(props)

    this.state = {
      animations: [
        !(this.props.activities[this.props.currentActivity].status && this.props.activities[this.props.currentActivity].status.reward < 1),
        !(this.props.activities[this.props.currentActivity].status && this.props.activities[this.props.currentActivity].status.reward < 2),
        !(this.props.activities[this.props.currentActivity].status && this.props.activities[this.props.currentActivity].status.reward < 3)
      ]
    }

    this.stars = []
  }

  componentDidMount () {
    /* istanbul ignore next */
    setTimeout(() => {
      if (this.state.animations[0]) {
        this.stars[0].zoomIn(1000).then(() => {
          if (this.state.animations[1]) {
            this.stars[1].zoomIn(1000).then(() => {
              if (this.state.animations[2]) {
                this.stars[2].zoomIn(1000)
              }
            })
          }
        })
      }
    }, 400)

    for (let i = 0; i < 3; i++) {
      /* istanbul ignore next */
      setTimeout(() => { if (this.state.animations[i]) this.stars[i].bounceOutDown(2000) }, 4000 + i * 400)
      /* istanbul ignore next */
      setTimeout(() => { if (this.state.animations[i]) this.props.increaseProgress(i + 1) }, 4400 + i * 400)
    }
  }

  render () {
    return (
      <View style={styles.rewardContainer}>
        <Animatable.Image
          ref={ref => (this.stars.push(ref))}
          style={styles.icon}
          resizeMode={'center'}
          source={Images.ui.star} />
        <Animatable.Image
          ref={ref => (this.stars.push(ref))}
          style={[styles.icon, styles.iconCenter]}
          resizeMode={'center'}
          source={Images.ui.star} />
        <Animatable.Image
          ref={ref => (this.stars.push(ref))}
          style={styles.icon}
          resizeMode={'center'}
          source={Images.ui.star} />
      </View>
    )
  }
}

RewardModalStars.propTypes = {
  activities: PropTypes.array.isRequired,
  currentActivity: PropTypes.number.isRequired,
  increaseProgress: PropTypes.func.isRequired
}
