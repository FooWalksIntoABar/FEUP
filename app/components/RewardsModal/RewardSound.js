import React from 'react'
import PropTypes from 'prop-types'
import { Audio } from 'expo'

export class RewardSound extends React.Component {
  sound = null

  componentDidMount () {
    setTimeout(this.playSound, 400)
  }

  componentWillUnmount () {
    if (this.sound !== null) {
      this.sound.unloadAsync()
    }
  }

  playSound = async () => {
    this.sound = new Audio.Sound()
    try {
      if (this.props.success) await this.sound.loadAsync(require('../../assets/sounds/happy.wav'))
      else await this.sound.loadAsync(require('../../assets/sounds/sad.wav'))
      this.sound.playAsync()
      this.sound.setPositionAsync(0)
    } catch (error) { }
  }

  render () {
    return null
  }
}

RewardSound.propTypes = {
  success: PropTypes.bool.isRequired
}
