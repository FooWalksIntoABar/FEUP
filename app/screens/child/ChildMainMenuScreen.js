import React, { Component } from 'react'
import { View, StatusBar, Image, TouchableOpacity } from 'react-native'
import { ScreenOrientation } from 'expo'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

import { toggleLevelUpModal } from '../../actions/childActions'
import { addRoutines } from '../../actions/gameActions'
import { ChildExperienceBar } from '../../components/ChildMainMenu/ChildExperienceBar'
import { LevelUpModal } from '../../components/ChildMainMenu/LevelUpModal'
import { Avatar } from '../../components/ChildMainMenu/Avatar'
import EnvVars from '../../constants/EnviromentVars'

import Images from '../../assets/images/images'

import styles from '../../styles/ChildMainMenuScreen.style'

export class ChildMainMenuScreen extends Component {
  constructor (props) {
    super(props)

    this.state = {
      isShopVisible: false,
      showModal: false,
      reward: { name: '', photo: undefined }
    }

    this.openShop = this.openShop.bind(this)
    this.closeShop = this.closeShop.bind(this)
    this.onCloseModal = this.onCloseModal.bind(this)
    this.showRewardModal = this.showRewardModal.bind(this)
  }

  componentDidMount () {
    ScreenOrientation.allow(ScreenOrientation.Orientation.LANDSCAPE)
  }

  componentWillUnmount () {
    ScreenOrientation.allow(ScreenOrientation.Orientation.PORTRAIT)
  }

  componentWillMount () {
    let url = `${EnvVars.apiUrl}routine_manager/get-daily-routine?selectedChildID=${this.props.childID}`
    fetch(url)
      .then((response) => response.json())
      .then((responseJson) => {
        if (responseJson.status === '200') {
          this.props.addRoutines(JSON.parse(responseJson.response))
        } else {

        }
        return responseJson
      })
      .catch((error) => {
        console.error(error)
      })
    this.fetchChildReward()
  }

  fetchChildReward = () => {
    let url = `${EnvVars.apiUrl}routine_manager/get-reward?selectedChildID=${this.props.childID}`
    fetch(url)
      .then((response) => response.json())
      .then((responseJson) => {
        if (responseJson.status === '200') {
          let reward = JSON.parse(responseJson.response).map(reward => ({ title: reward.name, ...reward }))[0]
          if (reward) {
            this.setState({ reward, showModal: this.props.showLevelUpModal })
          } else {
            this.setState({ reward: { name: '', photo: undefined } })
          }
        } else {

        }
        return responseJson
      })
      .catch((error) => {
        console.error(error)
      })
  }

  deleteReward (id) {
    fetch(EnvVars.apiUrl + 'routine_manager/remove-reward/', {
      method: 'DELETE',
      headers: {
        Accept: 'application/json',
        'Content-Type': 'application/json'
      },
      body: JSON.stringify({
        rewardID: id,
        childID: this.props.childID
      })
    }).then((response) => response.json())
      .then((responseJson) => {
        if (responseJson.status === '200') {
          console.log('trofeu apagada')
          this.fetchChildReward()
        } else {
          console.log('trofeu nao apagado')
        }
        return responseJson
      })
      .catch((error) => {
        console.error(error)
      })
  }

  openShop () {
    this.setState({ isShopVisible: true })
  }

  closeShop () {
    this.setState({ isShopVisible: false })
  }

  onCloseModal () {
    this.setState({ showModal: false })
    if (this.props.showLevelUpModal) {
      this.props.toggleLevelUpModal()
      this.deleteReward(this.state.reward.id)
    }
  }

  showRewardModal () {
    this.setState({ showModal: true })
  }

  render () {
    return (
      <View style={styles.mainMenuContainer}>
        <StatusBar hidden />
        <View style={styles.experienceBarContainer}>
          <ChildExperienceBar
            progress={(this.props.xp - this.props.level * 100) / 100}
            level={this.props.level}
            onPress={this.showRewardModal}
            photo={this.state.reward.photo} />
        </View>
        <View style={styles.buttonContainer}>
          <TouchableOpacity
            style={styles.profileButton}
            onPress={() => { this.props.navigation.navigate('ShopScreen') }}
            activeOpacity={0.9} >
            <Avatar button equiped={this.props.itemsEquiped} gender={this.props.gender} />
          </TouchableOpacity>
          <TouchableOpacity
            style={styles.button}
            onPress={() => { this.props.navigation.navigate('ChooseRoutineScreen') }}
            activeOpacity={0.9}>
            <Image style={styles.buttonImage} source={Images.ui.play} />
          </TouchableOpacity>
        </View>
        {this.state.showModal && <LevelUpModal
          level={this.props.level}
          xp={this.props.xp}
          isReward={this.props.showLevelUpModal}
          show={this.state.showModal}
          onClosed={this.onCloseModal}
          playSounds={this.props.playSounds}
          reward={this.state.reward} />}
      </View>
    )
  }
}

export default connect(
  /* istanbul ignore next */
  state => ({
    childID: state.child.id,
    level: state.child.level,
    xp: state.child.xp,
    gender: state.child.gender,
    itemsEquiped: state.child.itemsEquiped,
    showLevelUpModal: state.child.showLevelUpModal,
    playSounds: state.settings.playSounds
  }),
  /* istanbul ignore next */
  dispatch => ({
    toggleLevelUpModal: () => dispatch(toggleLevelUpModal()),
    addRoutines: routines => dispatch(addRoutines(routines))
  })
)(ChildMainMenuScreen)

ChildMainMenuScreen.propTypes = {
  navigation: PropTypes.object.isRequired,
  level: PropTypes.number.isRequired,
  xp: PropTypes.number.isRequired,
  showLevelUpModal: PropTypes.bool.isRequired,
  toggleLevelUpModal: PropTypes.func.isRequired,
  playSounds: PropTypes.bool.isRequired,
  gender: PropTypes.string.isRequired,
  itemsEquiped: PropTypes.array.isRequired,
  childID: PropTypes.number.isRequired,
  addRoutines: PropTypes.func.isRequired
}
