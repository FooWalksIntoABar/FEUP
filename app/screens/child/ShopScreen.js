import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { View, ScrollView, Image, Text } from 'react-native'
import { connect } from 'react-redux'

import Images, { avatarItems, avatarCategories } from '../../assets/images/images'
import { ShopItem } from '../../components/ChildMainMenu/ShopItem'
import { ShopTitle } from '../../components/ChildMainMenu/ShopTitle'
import { Avatar } from '../../components/ChildMainMenu/Avatar'

import styles from '../../styles/Shop.style'
import { purchaseItem, toggleItem } from '../../actions/childActions'

export class ShopScreen extends Component {
  render () {
    let shopItems = []

    for (let i = 0; i < avatarCategories.length; i++) {
      shopItems.push(<ShopTitle key={1000 + i} title={avatarCategories[i]} />)

      avatarItems
        .filter(item => item.category === avatarCategories[i])
        .forEach((item, index) => shopItems.push(
          <ShopItem
            key={i * 100 + index}
            id={item.id}
            cost={item.price}
            image={item.thumbnail}
            disabled={item.price > this.props.stars && !this.props.itemsOwned.includes(item.id)}
            purchaseItem={this.props.purchaseItem}
            toggleItem={this.props.toggleItem}
            equiped={this.props.itemsEquiped.includes(item.id)}
            purchased={this.props.itemsOwned.includes(item.id)} />
        ))
    }

    return (
      <View style={styles.shopModal}>
        <View style={styles.imageCard} >
          <Avatar equiped={this.props.itemsEquiped} gender={this.props.gender} />
        </View>
        <View style={styles.shopItemsContainer} >
          <View style={styles.shopTitleContainer}>
            <Text style={styles.shopTitle}>Loja</Text>
            <View style={styles.currentBalanceContainer}>
              <Image style={styles.currentBalanceIcon} resizeMode={'contain'} source={Images.ui.star} />
              <Text style={styles.currentBalance}>{this.props.stars}</Text>
            </View>
          </View>
          <ScrollView>
            <View style={styles.shopContainer}>
              {shopItems}
            </View>
          </ScrollView>
        </View>
      </View>
    )
  }
}

export default connect(
  state => ({
    stars: state.child.stars,
    itemsOwned: state.child.itemsOwned,
    itemsEquiped: state.child.itemsEquiped,
    gender: state.child.gender
  }),
  dispatch => ({
    purchaseItem: (cost, id) => dispatch(purchaseItem(cost, id)),
    toggleItem: id => dispatch(toggleItem(id))
  })
)(ShopScreen)

ShopScreen.propTypes = {
  stars: PropTypes.number.isRequired,
  gender: PropTypes.string.isRequired,
  itemsOwned: PropTypes.array.isRequired,
  purchaseItem: PropTypes.func.isRequired,
  toggleItem: PropTypes.func.isRequired,
  itemsEquiped: PropTypes.array.isRequired
}
