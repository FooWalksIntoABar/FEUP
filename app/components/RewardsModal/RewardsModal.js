import React from 'react'
import { View, Image, TouchableOpacity } from 'react-native'
import PropTypes from 'prop-types'

import Modal from 'react-native-modalbox'
import * as Animatable from 'react-native-animatable'
import Images from '../../assets/images/images'

import styles from '../../styles/RewardModal.style'
import { PastActivityIcons } from './PastActivityIcons'
import { RewardModalExperienceBar } from './RewardModalExperienceBar'
import { RewardModalStars } from './RewardModalStars'

export class RewardsModal extends React.Component {
  constructor (props) {
    super(props)

    this.state = {
      progress: (this.props.xp - this.props.level * 100) / 100
    }

    this.increaseProgress = this.increaseProgress.bind(this)
  }

  increaseProgress = (increase) => {
    this.setState({ progress: (this.props.xp - this.props.level * 100 + increase) / 100 })
  }

  returnURIorImage = (activity) => {
    // TODO: This checks whether the photo attribute is type URI and should probably just eventually be totally changed to URI.
    if (activity.photo !== undefined && activity.photo.includes('file://')) {
      return { uri: activity.photo }
    } else {
      return Images[activity.image]
    }
  }

  render () {
    let cardText = this.props.activities[this.props.currentActivity].status && this.props.activities[this.props.currentActivity].status.reward ? 'Atividade completada!' : 'Atividade falhada...'

    return (
      <Modal style={styles.rewardsModal} isOpen={this.props.activities[this.props.currentActivity].status !== undefined} backButtonClose={false} swipeToClose={false} backdropPressToClose={false}>
        <View style={styles.rewardContainer}>
          <Image style={[styles.icon, styles.iconGreyedOut]} resizeMode={'center'} source={Images.ui.star} />
          <Image style={[styles.icon, styles.iconGreyedOut, styles.iconCenter]} resizeMode={'center'} source={Images.ui.star} />
          <Image style={[styles.icon, styles.iconGreyedOut]} resizeMode={'center'} source={Images.ui.star} />
        </View>
        <RewardModalStars activities={this.props.activities} currentActivity={this.props.currentActivity} increaseProgress={this.increaseProgress} />
        <View style={styles.rewardCard}>
          <Animatable.Text
            animation={'fadeOut'}
            delay={2400}
            style={styles.cardTitle}>
            { cardText }
          </Animatable.Text>
          <RewardModalExperienceBar progress={this.state.progress} />
          <PastActivityIcons activities={this.props.activities} currentActivity={this.props.currentActivity} />
        </View>
        <View style={styles.buttonContainer}>
          <TouchableOpacity
            activeOpacity={0.8}
            style={[styles.button, styles.backButton]}
            onPress={this.props.backPress} >
            <Image
              style={styles.buttonImage}
              resizeMode={'contain'}
              source={Images.ui.cancel} />
          </TouchableOpacity>
          <TouchableOpacity
            activeOpacity={0.8}
            style={[styles.button, styles.nextButton]}
            onPress={this.props.nextPress} >
            <Image
              style={styles.buttonImage}
              resizeMode={'contain'}
              source={Images.ui.next} />
          </TouchableOpacity>
        </View>
      </Modal>
    )
  }
}

RewardsModal.propTypes = {
  activities: PropTypes.array.isRequired,
  currentActivity: PropTypes.number.isRequired,
  xp: PropTypes.number.isRequired,
  level: PropTypes.number.isRequired,
  nextPress: PropTypes.func.isRequired,
  backPress: PropTypes.func.isRequired
}
