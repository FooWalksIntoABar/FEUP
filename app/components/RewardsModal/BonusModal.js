import React from 'react'
import { View, Image, TouchableOpacity } from 'react-native'
import PropTypes from 'prop-types'

import * as Animatable from 'react-native-animatable'
import Images from '../../assets/images/images'

import styles from '../../styles/RewardModal.style'
import { RewardModalExperienceBar } from './RewardModalExperienceBar'
import { RewardModalStars } from './RewardModalStars'
import { RewardSound } from './RewardSound'

export class BonusModal extends React.Component {
  constructor (props) {
    super(props)

    this.state = {
      progress: (this.props.xp - this.props.level * 100) / 100
    }

    this.increaseProgress = this.increaseProgress.bind(this)
  }

  increaseProgress = (increase) => {
    this.setState({ progress: (this.props.xp - this.props.level * 100 + increase) / 100 })
  }

  render () {
    return (
      <View style={styles.rewardsModal}>
        <View style={styles.rewardContainer}>
          <Image style={[styles.icon, styles.iconGreyedOut]} resizeMode={'center'} source={Images.ui.star} />
          <Image style={[styles.icon, styles.iconGreyedOut, styles.iconCenter]} resizeMode={'center'} source={Images.ui.star} />
          <Image style={[styles.icon, styles.iconGreyedOut]} resizeMode={'center'} source={Images.ui.star} />
        </View>
        <RewardModalStars activities={[{ status: 3 }]} currentActivity={0} increaseProgress={this.increaseProgress} />
        <View style={styles.rewardCard}>
          <Animatable.Text
            animation={'fadeOut'}
            delay={2400}
            style={styles.cardTitle}>
            Rotina Completa
          </Animatable.Text>
          <RewardModalExperienceBar progress={this.state.progress} />
          <View style={styles.routineButtonContainer}>
            <TouchableOpacity
              activeOpacity={0.8}
              style={[styles.button, styles.nextButton]}
              onPress={this.props.nextPress} >
              <Image
                style={styles.buttonImage}
                resizeMode={'contain'}
                source={Images.ui.next} />
            </TouchableOpacity>
          </View>
          <RewardSound success playSounds={this.props.playSounds} />
        </View>
      </View>
    )
  }
}

BonusModal.propTypes = {
  xp: PropTypes.number.isRequired,
  level: PropTypes.number.isRequired,
  nextPress: PropTypes.func.isRequired,
  playSounds: PropTypes.bool.isRequired
}
