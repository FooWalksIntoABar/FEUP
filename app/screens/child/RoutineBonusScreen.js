import React, { Component } from 'react'
import PropTypes from 'prop-types'
import Confetti from 'react-native-confetti'
import { connect } from 'react-redux'
import { View } from 'react-native'

import { addStars } from '../../actions/childActions'
import { BonusModal } from '../../components/RewardsModal/BonusModal'
import { oppositeColor } from '../../styles/Colors'

export class RoutineBonusScreen extends Component {
  componentDidMount () {
    if (this._confettiView) setTimeout(() => this._confettiView.startConfetti(), 1000)
    if (this._confettiView2) setTimeout(() => this._confettiView2.startConfetti(), 1000)
    this.props.addStars(3)
  }

  componentWillUnmount () {
    if (this._confettiView) this._confettiView.stopConfetti()
    if (this._confettiView2) this._confettiView2.stopConfetti()
  }

  render () {
    return (
      <View style={{ backgroundColor: oppositeColor, height: '100%', width: '100%' }}>
        <View style={{ position: 'absolute', height: '100%', width: '50%' }}>
          <Confetti confettiCount={20} size={2} ref={ref => (this._confettiView = ref)} />
        </View>
        <View style={{ position: 'absolute', left: '50%', height: '100%', width: '50%' }}>
          <Confetti confettiCount={20} size={2} ref={ref => (this._confettiView2 = ref)} />
        </View>
        <BonusModal
          level={this.props.level}
          xp={this.props.xp}
          nextPress={() => this.props.navigation.popToTop()} />
      </View>
    )
  }
}

export default connect(
  /* istanbul ignore next */
  state => ({
    xp: state.child.xp,
    level: state.child.level
  }),
  /* istanbul ignore next */
  dispatch => ({
    addStars: stars => dispatch(addStars(stars))
  })
)(RoutineBonusScreen)

RoutineBonusScreen.propTypes = {
  navigation: PropTypes.object.isRequired,
  level: PropTypes.number.isRequired,
  xp: PropTypes.number.isRequired,
  addStars: PropTypes.func.isRequired
}
