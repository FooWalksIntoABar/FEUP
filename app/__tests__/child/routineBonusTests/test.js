import 'react-native'
import React from 'react'
import { configure, shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-16'

import { RoutineBonusScreen } from '../../../screens/child/RoutineBonusScreen'
import { BonusModal } from '../../../components/RewardsModal/BonusModal'

configure({ adapter: new Adapter() })

describe('RoutineBonusScreen snapshot', () => {
  it('renders RoutineBonusScreen correctly', () => {
    const wrapper = shallow(<RoutineBonusScreen
      navigation={{ navigate: jest.fn(), popToTop: jest.fn() }}
      level={1}
      xp={0}
      addStars={jest.fn()}Z
      playSounds />)
    expect(toJson(wrapper)).toMatchSnapshot()
    wrapper.find(BonusModal).at(0).props().nextPress()
    wrapper.unmount()
  })
})
