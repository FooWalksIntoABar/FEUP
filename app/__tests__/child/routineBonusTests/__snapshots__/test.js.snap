// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoutineBonusScreen snapshot renders RoutineBonusScreen correctly 1`] = `
<Component
  style={
    Object {
      "backgroundColor": "#FFA23D",
      "height": "100%",
      "width": "100%",
    }
  }
>
  <Component
    style={
      Object {
        "height": "100%",
        "position": "absolute",
        "width": "50%",
      }
    }
  >
    <ConfettiView
      confettiCount={20}
      size={2}
      timeout={30}
      untilStopped={false}
    />
  </Component>
  <Component
    style={
      Object {
        "height": "100%",
        "left": "50%",
        "position": "absolute",
        "width": "50%",
      }
    }
  >
    <ConfettiView
      confettiCount={20}
      size={2}
      timeout={30}
      untilStopped={false}
    />
  </Component>
  <BonusModal
    level={1}
    nextPress={[Function]}
    playSounds={true}
    xp={0}
  />
</Component>
`;
